src/fparser/two/utils.py

```diff
@@ -272,23 +272,52 @@ class Base(ComparableMixin):
                     is either str or FortranReaderBase.
       self.item   - Line instance (holds label) or None.
 
+    :param type cls: the class of object to create.
+    :param string: (source of) Fortran string to parse.
+    :type string: `str` or \
+                  :py:class:`fparser.common.readfortran.FortranReaderBase`
+    :param parent_cls: the parent class of this object.
+    :type parent_cls: `type`
+    :param parent: the parent of this node in the parse tree.
+    :type parent: :py:class:`fparser.two.utils.Base` or NoneType
+
     '''
     # This dict of subclasses is populated dynamically by code at the end
     # of this module. That code uses the entries in the
     # 'subclass_names' list belonging to each class defined in this module.
     subclasses = {}
 
+    def __init__(self, string, parent_cls=None, parent=None):
+        # pylint:disable=unused-argument
+        # During normal construction of the parse tree the parent information
+        # is set-up in the `init` method. However, it may also be provided
+        # to this constructor.
+        self.parent = parent
+
     @show_result
-    def __new__(cls, string, parent_cls=None):
-        """
-        Create a new instance of this object.
+    def __new__(cls, string, parent_cls=None, parent=None):
```

Comment:
Again, please remove `parent=None` if it is removed from `__init__`

Reply:
Done.